packages/cli/src/commands/lint.ts

```diff
@@ -34,15 +35,42 @@ const lintCommand: CommandModule = {
           return [(process.stdin as ReadStream & { fd: 0 }).fd];
         },
       })
-      .check((argv: Dictionary<unknown>) => {
-        if (argv.format !== void 0 && !(formatOptions as string[]).includes(String(argv.format))) {
-          throw new TypeError('Unspecified format');
+      .middleware((argv: Dictionary<unknown>) => {
+        const [format] = argv.format as string[] & { 0: string };
+        if (typeof argv.output === 'string') {
+          argv.output = { [format]: argv.output };
         }
-
+      })
+      .check((argv: Dictionary<unknown>) => {
         if (!Array.isArray(argv.documents) || argv.documents.length === 0) {
           throw new TypeError('No documents provided.');
         }
 
+        const format = argv.format as string[] & { 0: string };
+        const output = argv.output as Dictionary<unknown> | undefined;
+
+        if (format.length === 1) {
+          if (output === void 0 || Object.keys(output).length === 1) {
+            return true;
+          }
+
+          throw new TypeError('Output must be either string or unspecified when a single format is specified');
+        }
+
+        if (!isPlainObject(output)) {
+          throw new TypeError('Multiple outputs have to be provided when more than a single format is specified');
+        }
+
+        const keys = Object.keys(output);
+        if (format.length !== keys.length) {
```

Comment:
@P0lip looks great but if I understand correctly there is no way to have something like
`--format json --format html --output.json out.json`
to have one (or more) output(s) to file and one to stdout, right?

Reply:
Would be really cool to retain the `stdout` default